maths/gamma_recursive.py

```diff
@@ -0,0 +1,84 @@
+"""
+Gamma function is a very useful tool in physics.
+It helps calculating complex integral in a convenient way.
+for more info: https://en.wikipedia.org/wiki/Gamma_function
+"""
+
+# Importing packages
+from math import sqrt, pi
+from re import match
+from typing import Union
+
+
+def gamma(num : Union[int, float]) -> Union[int, float]:
+    """
+    Calculates the value of Gamma function of num
+    where num is either an integer (1,2,3..) or a half-integer (0.5,1.5,2.5...).
+    Implemented using recursion
+    Examples:
+    >>> Gamma of: 0.5
+    √π
```

Comment:
On your machine, please try `python3 -m doctest -v maths/gamma_recursive.py` to see if your doctests pass.

Reply:
Done.

Thanks